WordPress/src/androidTest/java/org/wordpress/android/support/WPSupportUtils.java

```diff
@@ -73,38 +80,72 @@ public static boolean isElementCompletelyDisplayed(ViewInteraction element) {
     public static void scrollToThenClickOn(Integer elementID) {
         waitForElementToBeDisplayed(elementID);
         onView(withId(elementID))
-                .perform(scrollTo())
-                .perform(click());
+                .perform(scrollTo());
+        clickOn(elementID);
     }
 
     public static void scrollToThenClickOn(ViewInteraction element) {
         waitForElementToBeDisplayed(element);
-        element.perform(scrollTo())
-                .perform(click());
+        element.perform(scrollTo());
+        clickOn(element);
     }
 
     public static void clickOn(Integer elementID) {
         waitForElementToBeDisplayed(elementID);
         clickOn(onView(withId(elementID)));
+        idleFor(500); // allow for transitions
     }
 
     public static void clickOn(ViewInteraction viewInteraction) {
         waitForElementToBeDisplayed(viewInteraction);
-        viewInteraction.perform(click());
-        idleFor(500);   // allow for transitions
+        idleFor(2000); // allow for transitions
+        viewInteraction.perform(click(closeSoftKeyboard())); // attempt to close the soft keyboard as the rollback
+        idleFor(500); // allow for transitions
     }
 
     /**
-     * Uses UIAutomator to click on an element using the resource ID in the cases of flakiness in Espresso click
-     * performing a long click
-     * @param resourceID - String resource ID
+     * Returns an action that performs a single click on the view.
+     *
+     * If the click takes longer than the 'long press' duration
+     * (which is possible because of
+     * https://android.googlesource.com/platform/frameworks/testing/+/android-support-test/espresso/core/src/main/
+     * java/android/support/test/espresso/action/GeneralClickAction.java#75)
+     * the provided rollback action is invoked on the view and a click is attempted again.
+     *
+     * In our case it triggers a toast so the rollback action will be
+     *
      */
-    public static void clickOn(String resourceID) {
+    public static ViewAction click(ViewAction rollbackAction) {
+        checkNotNull(rollbackAction);
+        return new GeneralClickAction(Tap.SINGLE,
+                GeneralLocation.CENTER,
+                Press.PINPOINT,
+                InputDevice.SOURCE_MOUSE,
+                MotionEvent.ACTION_DOWN,
+                rollbackAction);
+    }
+
+
+
+    private static boolean isResourceId(String text) {
+        return text.startsWith("id");
+    }
+
+    /**
+     * Uses UIAutomator to click on an element using either the resource ID or text in the cases of flakiness in
```

Comment:
The bots aren't happy about this one – the line is too long – can we constrain it to 120 chars?

Reply:
Weird 🤔 I checked and the length is 113 chars, I reduced it anyway to be on the safe side